x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityMigrationExecutor.java

```diff
@@ -46,10 +46,17 @@ public SecurityMigrationExecutor(
 
     @Override
     protected void nodeOperation(AllocatedPersistentTask task, SecurityMigrationTaskParams params, PersistentTaskState state) {
-        applyOutstandingMigrations(task, params.getMigrationVersion(), ActionListener.wrap((res) -> task.markAsCompleted(), (exception) -> {
+        ActionListener<Void> listener = ActionListener.wrap((res) -> task.markAsCompleted(), (exception) -> {
             logger.warn("Security migration failed: " + exception);
             task.markAsFailed(exception);
-        }));
+        });
+
+        if (params.isMigrationNeeded() == false) {
```

Comment:
can you also please log that the migration version has been bumped to latest without running them really?

Reply:
Yes, added a log message. Thanks!